polyfills/Array/from/polyfill.js

```diff
@@ -1,104 +1,205 @@
+		if (isNaN(number)) {
+			number = 0;
+		}
+		var len = (number >= 0 ? 1 : -1) * Math.floor(Math.abs(number));
+		if (len <= 0) { return 0; } // includes converting -0 to +0
+		if (len > MAX_SAFE_INTEGER) { return MAX_SAFE_INTEGER; }
+		return len;
+	}
+	function isCallable(value) {
+		return typeof value !== 'function' && typeof value !== 'object';
+	};
+	function isString (value) {
+		return typeof value === 'string';
+	}
+
+	var parseIterable = function (iterator) {
 		var done = false;
 		var iterableResponse;
 		var tempArray = [];
 
-		// if the iterable doesn't have next;
-		// it is an iterable if 'next' is a function but it has not been defined on
-		// the object itself.
-		if (typeof arraylike.next === 'function') {
+		if (iterator && typeof iterator.next === 'function') {
 			while (!done) {
-				iterableResponse = arraylike.next();
+				iterableResponse = iterator.next();
 				if (
-					iterableResponse.hasOwnProperty('value') &&
-					iterableResponse.hasOwnProperty('done')
+					iterableResponse.hasOwnProperty('value')
+					&& iterableResponse.hasOwnProperty('done')
 				) {
 					if (iterableResponse.done === true) {
 						done = true;
-						break;
+						break; // eslint-disable-line no-restricted-syntax
 
-					// handle the case where the done value is not Boolean
 					} else if (iterableResponse.done !== false) {
-						break;
+						break; // eslint-disable-line no-restricted-syntax
 					}
 
 					tempArray.push(iterableResponse.value);
+				} else if (iterableResponse.done === true) {
+					done = true;
+					break; // eslint-disable-line no-restricted-syntax
 				} else {
-
-					// it does not conform to the iterable pattern
-					break;
+					break; // eslint-disable-line no-restricted-syntax
 				}
 			}
 		}
 
-		if (done) {
-			return tempArray;
-		} else {
+		return done ? tempArray : false;
+	};
+
+	var hasSymbols = 'Symbol' in this && 'iterator' in Symbol;
+	var iteratorSymbol;
+	var forOf;
+	var hasSet = !!this.Set && isCallable(Set.prototype.values);
+	var hasMap = !!this.Map && isCallable(Map.prototype.entries);
+
+	if (hasSymbols) {
+		iteratorSymbol = Symbol.iterator;
+	} else {
+		var supportsStrIterator = (function () {
+			try {
+				var supported = false;
+				var obj = { // eslint-disable-line no-unused-vars
+					'@@iterator': function () {
+						return {
+							'next': function () {
+								supported = true;
+								return {
+									'done': true,
+									'value': undefined
+								};
+							}
+						};
+					}
+				};
+
+				Function('obj', // eslint-disable-line no-new-func
+					'for (var x of obj) {}'
+				)(obj);
+				return supported;
+			} catch (e) {
+				return false;
+			}
+		}());
 
-			// something went wrong return false;
-			return false;
+		if (supportsStrIterator) {
+			iteratorSymbol = '@@iterator';
+		} else {
+			try {
+				if (Function('var s = new Set(); s.add(0); for (var x of s) return x;')() === 0) { // eslint-disable-line no-new-func
```

Comment:
Agree. We should avoid using the `new Function` form if possible. There was an issue some time ago to remove all existing instances.

However, it's ok if the polyfill is only required in browsers that also don't support CSP. This should be marked with a comment.